gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/revocation/impl/RevocationTokenServiceImpl.java

```diff
@@ -110,10 +120,14 @@ public Completable revoke(RevocationTokenRequest request, Client client) {
                     // Log the result anyway for posterity.
                     if (throwable instanceof InvalidTokenException) {
                         logger.debug("No refresh token {} found in the token store.", token);
+                        auditService.report(AuditBuilder.builder(ClientTokenAuditBuilder.class).revoked());
                         return Completable.complete();
                     }
                     return Completable.error(throwable);
-                });
+                })
+                .doOnError(error -> auditService.report(AuditBuilder.builder(ClientTokenAuditBuilder.class)
+                        .throwable(error)
```

Comment:
There is no access to tokenId here and this doOnError could be moved 149 and 167 lines but in that case, it will be produce 2 logs - one will be incorrect.

Reply:
let's me think about it, I need to checkout the branch and play with it.

Reply:
ok. There are tests for that case. Please look at them.